src/components/Tooltip/Tooltip.tsx

```diff
@@ -266,6 +258,14 @@ export const Tooltip = ({
         // eslint-disable-next-line react-hooks/exhaustive-deps
     }, [isOpen]);
 
+    useEffect(() => {
+        window.addEventListener('keydown', (event) => {
+            if (isOpen && event.key === 'Escape') {
+                setIsOpen(false);
+            }
+        });
+    });
+
```

Comment:
I would put the callback into a constant and then removing the event from the window object for that callback. Otherwise, a new event will be added every time the component renders and all of them will be called when pressing the Esc key.

```typescript
useEffect(() => {
  window.addEventListener('keydown', handleCloseTooltipOnEscape);

  return () => {
    window.removeEventListener('keydown', handleCloseTooltipOnEscape);
  }
}, []);
```

Reply:
Added